server/image_pull.go

```diff
@@ -49,6 +83,17 @@ func (s *Server) PullImage(ctx context.Context, req *types.PullImageRequest) (*t
 		}
 	}
 
+	rhName, rh := s.GetRuntimeHandlerForPod(ctx, req.SandboxConfig)
+	if rh != nil {
+		if rh.RuntimePullImage {
+			// don't pull the image in crio, as it will be done by the runtime
+			log.Debugf(ctx, "Skip image pull for runtime %s - image %s", rhName, image)
+			return &types.PullImageResponse{
+				ImageRef: req.Image.Image,
```

Comment:
Nowadays, the returned `ImageRef` is always a digested reference. If this code path starts putting less structured user-controlled data into the field, I don’t immediately know what that means. E.g. does `ImageStatus` accept these values?